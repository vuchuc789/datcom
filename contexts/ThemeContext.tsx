'use client';

import {
  createTheme,
  CssBaseline,
  ThemeProvider as MuiThemeProvider,
} from '@mui/material';
import { useMemo } from 'react';

type ThemeProviderProps = React.PropsWithChildren;

export const ThemeProvider: React.FC<ThemeProviderProps> = ({ children }) => {
  const theme = useMemo(() => createTheme(), []);

  return (
    <MuiThemeProvider theme={theme}>
      <CssBaseline />
      {children}
    </MuiThemeProvider>
  );
};
