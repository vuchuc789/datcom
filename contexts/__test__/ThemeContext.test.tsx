import { render } from '@testing-library/react';
import { ThemeProvider } from '../ThemeContext';

describe('ThemeProvider', () => {
  test('render correctly', () => {
    const { container } = render(<ThemeProvider>sample content</ThemeProvider>);

    expect(container).toMatchSnapshot();
  });
});

export {};
