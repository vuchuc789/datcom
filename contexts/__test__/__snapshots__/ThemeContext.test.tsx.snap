// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeProvider render correctly 1`] = `
<div>
  sample content
</div>
`;
